src/Claim.ts

```diff
@@ -1,42 +1,62 @@
 /* tslint:disable:no-relative-imports */
 import * as bitcore from 'bitcore-lib'
 import * as crypto from 'crypto'
-
+import { canonize, expand } from 'jsonld'
 import { IllegalArgumentException } from './Exceptions'
 import { Claim, ClaimAttributes, ClaimType, isClaim } from './Interfaces'
-import { Serialization } from './Serialization'
 
-export function getClaimId(claim: Claim): string {
-  const buffer = Buffer.from(
-    Serialization.claimToHex({
-      ...claim,
-      id: '',
-      signature: '',
-    }),
-    'hex'
-  )
-  return crypto
-    .createHash('sha256')
-    .update(buffer)
-    .digest()
-    .toString('hex')
+const claimContex = {
+  attributes: 'http://schema.org/CreativeWork',
+  author: 'http://schema.org/author',
+  text: 'http://schema.org/text',
+  created: 'http://purl.org/dcterms/created',
+  dateCreated: 'http://schema.org/dateCreated',
+  datePublished: 'http://schema.org/datePublished',
+  name: 'http://schema.org/name',
+  tags: 'http://schema.org/keyword',
+  type: 'http://schema.org/additionalType',
+  publicKey: 'http://schema.org/Text',
+}
+
+export const expandClaim = async (claim: Claim): Promise<any> => {
+  const contextualClaim = { '@context': claimContex, ...claim }
+  return expand(contextualClaim)
+}
+
+export const canonizeClaim = async (claim: Claim): Promise<string> => {
+  return canonize(await expandClaim(claim))
 }
 
-export function getClaimSignature(claim: Claim, privateKey: string): string {
+export const getClaimId = async (claim: Claim): Promise<string> => {
+  try {
```

Comment:
I think this try/catch is not doing anything...

Reply:
ack